// @react-email/section/dist/index.mjs
// src/section.tsx
import * as React from "react";
import { jsx } from "react/jsx-runtime";
var Section = React.forwardRef(
  ({ children, style, ...props }, forwardedRef) => {
    const styleDefaultTable = {
      width: "100%",
      ...style
    };
    const styleDefaultTr = {
      display: "grid",
      gridAutoColumns: "minmax(0, 1fr)",
      gridAutoFlow: "column"
    };
    const arrayChildren = React.Children.toArray(children);
    const hasTdElement = (child) => {
      return React.isValidElement(child) && child.type === "td";
    };
    const finalChildren = arrayChildren.map((child, index) => {
      return hasTdElement(child) ? child : /* @__PURE__ */ jsx("td", { children: child }, index);
    });
    return /* @__PURE__ */ jsx(
      "table",
      {
        ref: forwardedRef,
        style: styleDefaultTable,
        align: "center",
        border: 0,
        cellPadding: 0,
        cellSpacing: 0,
        role: "presentation",
        ...props,
        children: /* @__PURE__ */ jsx("tbody", { children: /* @__PURE__ */ jsx("tr", { style: styleDefaultTr, children: finalChildren }) })
      }
    );
  }
);
Section.displayName = "Section";
export {
  Section
};
